newton/_src/usd/utils.py

```diff
@@ -0,0 +1,459 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 The Newton Developers
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from collections.abc import Iterable, Sequence
+from typing import Any, overload
+
+import numpy as np
+import warp as wp
+
+from ..core.types import Axis, AxisType, nparray
+from ..sim.model import CustomAttribute, ModelAttributeAssignment, ModelAttributeFrequency
+
+try:
+    from pxr import Gf, Usd, UsdGeom
+except ImportError:
+    Usd = None
+    Gf = None
+    UsdGeom = None
+
+
+def get_attribute(prim: Usd.Prim, name: str, default: Any | None = None) -> Any | None:
+    """
+    Get an attribute value from a USD prim, returning a default if not found.
+
+    Args:
+        prim: The USD prim to query.
+        name: The name of the attribute to retrieve.
+        default: The default value to return if the attribute is not found or invalid.
+
+    Returns:
+        The attribute value if it exists and is valid, otherwise the default value.
+    """
+    attr = prim.GetAttribute(name)
+    if not attr or not attr.HasAuthoredValue() or not attr.IsValid():
```

Comment:
I know its weird but `not attr` and `not attr.IsValid()` are actually the same check.